src/alert-manager/src/alert-handler/controllers/node.js

```diff
@@ -72,7 +73,144 @@ const cordonNodes = (req, res) => {
+            {
+              name: 'nvidia-gpu-low-perf-fixer',
+              image: `${DOCKER_REGISTRY_PREFIX}nvidia-gpu-low-perf-fixer:${DOCKER_REGISTRY_TAG}`,
+              imagePullPolicy: 'Always',
+              env: [
+                {
+                  name: 'MINOR_NUMBER',
+                  value: `${minorNumber}`,
+                },
+              ],
+              securityContext: {
+                privileged: true,
+              },
+            },
+          ],
+          restartPolicy: 'Never',
+          nodeSelector: {
+            'kubernetes.io/hostname': nodeName,
+          },
+        },
+      },
+    },
+  };
+  return job;
+};
+
+// start a k8s job for each GPU card to fix NvidiaGPULowPerf issue
+const fixNvidiaGPULowPerf = (req, res) => {
+  logger.info(
+    'Received `fixNvidiaGPULowPerf` post request from alert-manager.',
+  );
+  // filter alerts which are firing and contain `node_name` & `minor_number` as label
+  const jobsInfo = req.body.alerts
+    .filter(
+      (alert) =>
+        alert.status === 'firing' &&
+        'node_name' in alert.labels &&
+        'minor_number' in alert.labels,
+    )
+    // map each alert to a job
+    .map((alert) => ({
+      jobName: `nvidia-gpu-low-perf-fixer-${crypto
+        .createHash('md5')
+        .update(alert.labels.node_name + alert.labels.minor_number)
+        .digest('hex')}`, // unique job by GPU card
+      nodeName: alert.labels.node_name,
+      minorNumber: alert.labels.minor_number,
+      DOCKER_REGISTRY_PREFIX: process.env.DOCKER_REGISTRY_PREFIX,
+      DOCKER_REGISTRY_TAG: process.env.DOCKER_REGISTRY_TAG,
+    }));
+
+  const k8sApi = kc.makeApiClient(k8s.BatchV1Api);
+  jobsInfo.forEach(async (jobInfo) => {
+    // get k8s V1Job
+    const job = getK8sV1Job(
+      jobInfo.jobName,
+      jobInfo.nodeName,
+      jobInfo.minorNumber,
+    );
+    k8sApi
+      .createNamespacedJob('default', job)
+      .then((response) => {
+        logger.info(
+          `Successfully start job ${jobInfo.jobName} for GPU Low Performance issue in node: ${jobInfo.nodeName}, minor number: ${jobInfo.minorNumber}`,
+        );
+      })
+      .catch((error) => {
+        // ignore the job creation if already exists
+        if (error.response && error.response.statusCode === 409) {
+          logger.warn(`Kubernetes job ${jobInfo.jobName} already exists.`);
+        } else {
+          logger.error(error);
+          res.status(500).json({
+            message: `Failed to start job to fix NvidiaGPULowPerf`,
+          });
+        }
+      });
+  });
+};
+
+// clean completed jobs which were used to fix NvidiaGPULowPerf issue
+// the jobs completed for more than 24 hours will be deleted
+const cleanCompletedfixNvidiaGPULowPerfJobs = (req, res) => {
+  logger.info(
+    'Cleaning completed jobs which were used to fix NvidiaGPULowPerf issue...',
+  );
+
+  const k8sApi = kc.makeApiClient(k8s.BatchV1Api);
+  k8sApi
+    .listNamespacedJob('default')
```

Comment:
You don't need to get all jobs here. You can add label to the job and list them by labels. Which is more efficient

Reply:
> You don't need to get all jobs here. You can add label to the job and list them by labels. Which is more efficient

Added a label `time-to-live=24h`